fluent-react/src/localized.ts

```diff
@@ -1,34 +1,28 @@
-import { isValidElement, cloneElement, useContext } from "react";
+import {
+  Fragment,
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createElement,
+  isValidElement,
+  useContext
+} from "react";
 import PropTypes from "prop-types";
-import FluentContext from "./context";
-import VOID_ELEMENTS from "../vendor/voidElementTags";
+import { FluentContext } from "./context";
+import { VOID_ELEMENTS } from "./vendor/voidElementTags";
+import { FluentArgument } from "@fluent/bundle";
 
 // Match the opening angle bracket (<) in HTML tags, and HTML entities like
 // &amp;, &#0038;, &#x0026;.
 const reMarkup = /<|&#?\w+;/;
 
-/*
- * Prepare props passed to `Localized` for formatting.
- */
-function toArguments(props) {
-  const args = {};
-  const elems = {};
-
-  for (const [propname, propval] of Object.entries(props)) {
-    if (propname.startsWith("$")) {
-      const name = propname.substr(1);
-      args[name] = propval;
-    } else if (isValidElement(propval)) {
-      // We'll try to match localNames of elements found in the translation with
-      // names of elements passed as props. localNames are always lowercase.
-      const name = propname.toLowerCase();
-      elems[name] = propval;
-    }
-  }
-
-  return [args, elems];
+interface LocalizedProps {
+  id: string;
+  attrs?: Record<string, boolean>;
+  children?: ReactNode;
+  vars?: Record<string, FluentArgument>;
```

Comment:
Just out of curiosity: why `vars` instead of `args`?

Reply:
Hmm, I'm torn on this one. They're called [_variables_](https://projectfluent.org/fluent/guide/variables.html) in Fluent Syntax, but they're passed as _arguments_. In DOM, we use `data-l10n-args`, too, but that's mostly an artifact of how we did it in Firefox OS. I think `vars` is a more accurate name here, but `args` might be a better choice because of the prior work and due to how developers think about them.

Reply:
@Pike, @flodolo: do you have any feedback about naming here? We're considering `vars` vs. `args`, that is:

    <Localized id="hello" vars={{userName: "you"}}/>

vs.

    <Localized id="hello" args={{userName: "you"}}/>

Reply:
My main concern would be to be consistent, and I'd rather change the resolvers as we change resolvers anyway. As opposed to changing the syntax.

So, `vars` is my vote.

One  could argue that you pass your variables as args. As in, when you pass them, they're args, when you store and reference them, they're variables, but that's not useful for anything, I think. (But that may be  the historical reason why we have both.)